src/Workspaces/Core/Portable/Workspace/Host/SyntaxTreeFactory/AbstractSyntaxTreeFactoryService.AbstractRecoverableSyntaxRoot.cs

```diff
@@ -62,7 +62,7 @@ internal SyntaxTreeInfo WithOptionsAndLength(ParseOptions options, int length)
         internal sealed class RecoverableSyntaxRoot<TRoot> : RecoverableWeakValueSource<TRoot>
             where TRoot : SyntaxNode
         {
-            private ITemporaryStreamStorage _storage;
+            private volatile ITemporaryStreamStorage _storage;
```

Comment:
Please don't use volatile.  This is becoming complex in a manner that has not been given justificaiton.